kong/templates/nginx.lua

```diff
@@ -23,17 +23,44 @@ events {
 > end
 }
 
-> if wasm and wasm_modules_parsed and #wasm_modules_parsed > 0 then
+> if wasm then
+load_module $(wasm_dynamic_module);
+
 wasm {
+  env RUST_BACKTRACE=1;
+  env WASMTIME_BACKTRACE_DETAILS=1;
   shm_kv kong_wasm_rate_limiting_counters 12m;
-
+> for _, el in ipairs(wasm_main_shm_directives) do
+  shm_kv $(el.name) $(el.value);
+> end
 > for _, module in ipairs(wasm_modules_parsed) do
   module $(module.name) $(module.path);
 > end
+> for _, el in ipairs(wasm_main_directives) do
+  $(el.name) $(el.value);
+> end
```

Comment:
We'll need to filter out items that start with `wasm_` but aren't intended for nginx.conf at some level. When I run integration tests locally I get errors because the values of `wasm_dynamic_module` and `wasm_filters_path` are treated as injected directives.

Not sure what the best approach would be for this. We could...

* maintain a small list of known items to filter out
* change the prefix of the injected directives to something like `nginx_wasm_*`
* change the prefix on the _non_ nginx directives to something else, like instead of `wasm_filters_path` make it `wasmx_filters_path`

Reply:
I have changed all directives prefix to `nginx_wasm_` - it's more aligned with the existing injected directives and resolves the issue here.